### PR TITLE
Custom Table Styling options

### PR DESCRIPTION
**Pull request by panzerox123**
# Description

Adds some fields to the custom Table widget that allows control over the styling. 
- New type `CustomColColor` that stores column number and color type.
- `ColColor` array. Appending to this array with {ColNumber: <int>, ColColor: <ui.Color>} will add custom colors to specified columns.
- `HeaderStyle` to allow changing of header style
- `RowStyle` to allow changing the default row style

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [X] I have read the [contribution guidelines](https://github.com/pesos/grofer/blob/master/CONTRIBUTING.md) and followed it as far as possible. 
- [X] I have performed a self-review of my own code (if applicable)
- [X] I have commented my code, particularly in hard-to-understand areas (if applicable)
- [X] I have run `go fmt` on my code ([reference](https://blog.golang.org/gofmt))
- [ ] I have made corresponding changes to the documentation (if applicable)
- [X] My changes generate no new warnings 
- [ ] Any dependent and pending changes have been merged and published

### CODE REVIEW

File: src/utils/table.go [hunk 1]
```diff
@@ -25,13 +25,22 @@ import (
 	ui "github.com/gizak/termui/v3"
 )
 
+type CustomColColor struct {
+	ColNumber int
+	ColColor  ui.Color
+}
+
```
Comment (Gituser143) [inline src/utils/table.go:32]: @panzerox123 would it make sense to turn this into a map instead? It would simplify the colour setting within the `Draw()` function too.

**Review by Gituser143 — CHANGES_REQUESTED** [review]

File: src/utils/table.go (hunk 1 above)
Comment (panzerox123) [reply src/utils/table.go:32]: That makes sense. I'll give it a try

File: src/display/process/init.go [hunk 2]
```diff
@@ -188,7 +188,10 @@ func (page *AllProcPage) InitAllProc() {
 	}
 	page.ProcTable.ShowCursor = true
 	page.ProcTable.CursorColor = ui.ColorCyan
+	page.ProcTable.RowStyle = ui.NewStyle(ui.ColorClear)
+	page.ProcTable.ColColor[1] = ui.ColorGreen
 	page.ProcTable.BorderStyle.Fg = ui.ColorCyan
+	//page.ProcTable.HeaderStyle = ui.NewStyle(ui.ColorBlue, ui.ColorClear, ui.ModifierUnderline)
```
Comment (Gituser143) [inline src/display/process/init.go:194]: Could you remove this comment?

**Review by Gituser143 — CHANGES_REQUESTED** [review]
Apart from the single comment, everything else looks good! I'll try testing out some weird cases locally before approving and merging.

File: src/display/process/init.go (hunk 2 above)
Comment (panzerox123) [reply src/display/process/init.go]: Done!

**Comment by panzerox123** [comment]
> Apart from the single comment, everything else looks good! I'll try testing out some weird cases locally before approving and merging.

Cool!

**Comment by Gituser143** [comment]
image

@panzerox123 Maybe remove the background highlight? Looks kinda weird.

**Comment by Gituser143** [comment]
Screenshot from 2021-05-26 15-26-42

@panzerox123 Maybe remove the background highlight? Looks kinda weird I feel...

**Comment by panzerox123** [comment]
> @panzerox123 Maybe remove the background highlight? Looks kinda weird I feel...

My only concern is that it can get hard to read
image

**Comment by Gituser143** [comment]
> > @panzerox123 Maybe remove the background highlight? Looks kinda weird I feel...
> 
> My only concern is that it can get hard to read
> image

Wait can't it be made same colour as rest of row when selected?

**Comment by panzerox123** [comment]
image
I believe this is what you meant?

**Comment by Gituser143** [comment]
> I believe this is what you meant?
Yes! Looks great!

File: src/display/process/allProcs.go [hunk 3]
```diff
@@ -296,7 +296,7 @@ func AllProcVisuals(dataChannel chan []*proc.Process,
 
 		case data := <-dataChannel:
 			if runAllProc {
-				myPage.ProcTable.CursorColor = selectedStyle
+				//myPage.ProcTable.CursorColor = selectedStyle
```
Comment (Gituser143) [inline src/display/process/allProcs.go:299]: Wait, why was this removed?

**Review by Gituser143 — CHANGES_REQUESTED** [review]

File: src/display/process/allProcs.go (hunk 3 above)
Comment (panzerox123) [reply src/display/process/allProcs.go:299]: My bad I didn't save the file before committing.

**Review by Gituser143 — APPROVED** [review]

**Review by MadhavJivrajani — APPROVED** [review]
Lgtm, going with @Gituser143's judgement here 

Merge whenever time permits @Gituser143 :)
